File: C++/1931.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(pair<int, int> _a, pair<int, int> _b)
{
    if (_a.second == _b.second) { return _a.first < _b.first; }
    return _a.second < _b.second;
}

int main()
{
    int answer = 0;
    int rooms;
    int start, end;
    int currentTime = 0;
    cin >> rooms;

    vector<pair<int, int>> meetingTime(rooms);

    for (int idx = 0; idx < meetingTime.size(); idx++)
    {
        cin >> meetingTime[idx].first >> meetingTime[idx].second;
    }

    sort(meetingTime.begin(), meetingTime.end(), compare);

    for (int idx = 0; idx < rooms; idx++)
    {
        if (meetingTime[idx].first >= currentTime)
        {
            currentTime = meetingTime[idx].second;
            answer++;
        }
    }
    cout << answer;
    return 0;
}

File: C#/1931.cs
using System;
using System.Linq;

namespace ConsoleApp1
{
    class Program
    {
        static void Main()
        {
            int answer = 0;
            int rooms = int.Parse(Console.ReadLine());

            int currentTime = 0;

            (int start, int end)[] meetingTime = new (int, int)[rooms];

            for (int idx = 0; idx < rooms; idx++)
            {
                string timeString = Console.ReadLine();
                string[] InputTimes = timeString.Split();

                meetingTime[idx].start = int.Parse(InputTimes[0]);
                meetingTime[idx].end = int.Parse(InputTimes[1]);
            }

            meetingTime = meetingTime.OrderBy(x => x.end).ThenBy(x => x.start).ToArray();

            for (int idx = 0; idx < rooms; idx++)
            {
                if (meetingTime[idx].start >= currentTime)
                {
                    currentTime = meetingTime[idx].end;
                    answer++;
                }
            }

            Console.WriteLine(answer);
        }
    }
}